Source/SwiftLintFramework/Models/Configuration.swift

```diff
@@ -78,14 +76,17 @@ public struct Configuration: Equatable {
         }
         self.disabledRules = validDisabledRules
 
-        self.rules = rules.filter {
-            !validDisabledRules.contains($0.dynamicType.description.identifier)
+        self.rules = rules.filter { rule in
+            let id = rule.dynamicType.description.identifier
+            if validDisabledRules.contains(id) { return false }
+            return enabledRules.contains(id) || !(rule is OptInRule)
```

Comment:
I get it, although I got a different meaning from @scottrhoyt 's comment which seems to imply either a white list or black list, but not both (or opt in).